File: app/src/main/java/com/util/base/MainActivity.java
package com.util.base;

import com.common.base.binding.BindingActivity;
import com.util.base.databinding.ActivityMainBinding;

public class MainActivity extends BindingActivity<ActivityMainBinding> {

    @Override
    protected int getContentViewId() {
        return R.layout.activity_main;
    }

    @Override
    public ActivityMainBinding getViewBinding() {
        return ActivityMainBinding.inflate(getLayoutInflater());
    }

    @Override
    protected void initData() {

    }
}

File: app/src/main/java/com/util/base/MainActivity.kt
package com.util.base

import com.common.base.mvp.MvpActivity
import com.util.base.MainContract.Presenter

class MainActivity : MvpActivity<Presenter>(), MainContract.View {

    override fun getContentViewId(): Int {
        return R.layout.activity_main
    }

    override fun createPresenter(): Presenter {
        return MainPresenter(this, this)
    }

    override fun initData() {
        mPresenter?.refreshToken()
    }


    override fun showTips(message: String) {}


}